### PR TITLE
sw_engine raster: code refactoring.

### PR DESCRIPTION
**Pull request by hermet**
Separate simd implementation by files to maintain them easier.

Now we have avx, c, neon version implementation base.
we can add implementations to them.

### CODE REVIEW

File: src/lib/sw_engine/tvgSwRasterNeon.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        uint8x8_t vSrc = (uint8x8_t) vdup_n_u32(src);
+        uint8x8_t vIalpha = (uint8x8_t) vdup_n_u32(ialpha);
+
+        uint32_t iterations = span->len / 2;
+        uint32_t left = span->len % 2;
+
+        for (uint32_t x = 0; x < iterations; x+=2) {
+            vDst[x] = vadd_u8(vSrc, ALPHA_BLEND_NEON(vDst[x], vIalpha));
+        }
```
Comment (mgrudzinska) [inline src/lib/sw_engine/tvgSwRasterNeon.h:68]: Is the formatting consistent with the tvg convention?

File: src/lib/sw_engine/tvgSwRasterNeon.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        if (left) {
+            dst[span->len] = src + ALPHA_BLEND(dst[span->len], ialpha);
+        }
```
Comment (mgrudzinska) [inline src/lib/sw_engine/tvgSwRasterNeon.h:72]: Is the formatting consistent with the tvg convention?

**Review by mgrudzinska — APPROVED** [review]
@hermet besides some questions about formatting I approve

**Review by mihashco — APPROVED** [review]

File: src/lib/sw_engine/tvgSwRasterNeon.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    uint32_t src;
+
+    for (uint32_t i = 0; i < rle->size; ++i) {
+        auto dst = &surface->buffer[span->y * surface->stride + span->x];
+
+        uint8x8_t *vDst = (uint8x8_t*) dst;
+
+        if (span->coverage < 255) src = ALPHA_BLEND(color, span->coverage);
+        else src = color;
+        auto ialpha = 255 - surface->blender.alpha(src);
```
[the remark this answers is not in the document]
Comment (hermet) [reply src/lib/sw_engine/tvgSwRasterNeon.h:58]: Coding convention of the original code is out of this refactoring and arguing preference is non-sense.

File: src/lib/sw_engine/tvgSwRasterNeon.h (hunk 1 above)
Comment (hermet) [reply src/lib/sw_engine/tvgSwRasterNeon.h:68]: Hmm it looks no problem to me. what would you expect to be here? can you explain more detail?

File: src/lib/sw_engine/tvgSwRasterNeon.h (hunk 2 above)
Comment (hermet) [reply src/lib/sw_engine/tvgSwRasterNeon.h:72]: Hmm it looks no problem to me. what would you expect to be here? can you explain it more detail?

**Pull request merged**